plugins/edgeos.sh

```diff
@@ -98,13 +98,15 @@ edgeos_power() {
 edgeos_temperature() {
     temperature_info=$(sudo /usr/sbin/ubnt-hal getTemp | tail -n +2 | head -n -1)
 
-    if ! echo "$temperature_info" | grep "is not supported on this platform"; then
+    if ! [ -z "$temperature_info" ]; then
         echo "$temperature_info" | while read -r line; do
             temperature_info_sensor=$(echo "$line" | cut -d ':' -f 1 | sed 's/ /\\ /')
             temperature_info_value=$(echo "$line" | cut -d ':' -f 2 | cut -d ' ' -f 1)
 
             echo "edgeos_temperature,sensor=$temperature_info_sensor temperature=$temperature_info_value"
         done
+    else
+        echo "Temperature not supported on this platform"
```

Comment:
I'm not quite sure what you mean -- the `else` condition should only run if `getTemp` is not supported. Therefore, it would behave as the `getPowerStatus` command does, which is echoing the error message instead of the line protocol, intentionally resulting in an error in Telegraf. 

I was just trying to be consistent with the `getPowerStatus` behaviour; if you'd rather neither throw an error in Telegraf then these lines can be deleted, and the `-q` parameter should be used in the `grep` statement of `getPowerStatus` to suppress the error.

Reply:
Okay, so maybe we should think about the default behavior. What should happen if we enable an unsupported metric?

* `print nothing`: This would sot break something, even if you enable it
* `print an error`: This would fill your error logs, but that is maybe what we want?

I am ok with booth options. What would you prefer?

Unfortunately I have only a small amount of devices here to check things. The supported/unsupported stuff can only be done together with others. So I really appreciate your commitment.

Reply:
I would personally prefer that an error is printed. If I enabled a metric without checking the comparability chart first, and was wondering why I wasn't seeing that metric in my database, the first thing that I would do is check the logs. There isn't any reason, other than a misconfiguration, where it would fill your error logs. Silent failures are hard to trace, so I'd prefer to log the misconfiguration somewhere.